packages/components/src/components/domainproperties/dataset/DatasetPropertiesPanelFormElements.tsx

```diff
@@ -0,0 +1,280 @@
+                            label={"Category"}
+                            required={false}
+                            helpTipBody={() => this.getHelpTipElement("category")}
+                        />
+                    </Col>
+
+                    <Col xs={7}>
+                        {/*TODO: could be replaced by QuerySelect along with server side story*/}
+                        <Creatable
+                            name="categoryId"
+                            placeholder="Select dataset category"
+                            onChange={onCategoryChange}
+                            value={model.categoryId}
+                            options={availableCategories}
+                        />
+                    </Col>
+
+                    <Col xs={1}/>
+
+                </Row>
+
+                <DatasetSettingsInput
+                    name="label"
+                    label="Label"
+                    helpTip={this.getHelpTipElement("label")}
+                    value={model.label}
+                    disabled={false}
+                    onValueChange={onInputChange}
+                    showInAdvancedSettings={false}
+                    required={false}
+                />
+
+            </>
+        );
+    }
+}
+
+interface DataRowUniquenessElementsProps {
+    onRadioChange: (evt: any) => any;
+    dataRowSetting: number;
+}
+
+class DataRowUniquenessElements extends React.PureComponent<DataRowUniquenessElementsProps> {
+    render() {
+        const { onRadioChange, dataRowSetting } = this.props;
+        const radioName = "dataRowSetting";
+
+        return (
+            <div className='dataset_data_row_uniqueness_container'>
+                <Radio
+                    name={radioName}
+                    value={0}
+                    checked={dataRowSetting == 0}
+                    onChange={onRadioChange}
+                >
+                    Participant ID only (demographic data)
+                </Radio>
+
+                <Radio
+                    name={radioName}
+                    value={1}
+                    checked={dataRowSetting == 1}
+                    onChange={onRadioChange}
+                >
+                    Participant ID and timepoint
+                </Radio>
+
+                <Radio
+                    name={radioName}
+                    value={2}
+                    checked={dataRowSetting == 2}
+                    onChange={onRadioChange}
+                >
+                    Participant ID, timepoint, and additional key field
+                </Radio>
+            </div>
+        );
+    }
+}
+
+interface DataRowUniquenessContainerProps {
+    model: DatasetModel;
+    onRadioChange: (e: any) => any;
+    onCheckBoxChange: (any) => void;
+    onSelectChange: (name, formValue, selected) => void;
+}
+
+export class DataRowUniquenessContainer extends React.PureComponent<DataRowUniquenessContainerProps> {
+
+    getDataRowSetting(model: DatasetModel) : number {
```

Comment:
Since this function only uses the information from the DatasetModel object we could move this to the model object directly so it would be a call to model.getDataRowSetting()